src/atcore.cpp

```diff
@@ -150,8 +151,22 @@ QList<QSerialPortInfo> AtCore::serialPorts() const
 void AtCore::newMessage(const QByteArray &message)
 {
     lastMessage = message;
+    if (message.startsWith(QString::fromLatin1("X:").toLocal8Bit())) {
+        posString = message;
+        posString.resize(posString.indexOf("E"));
+        posString.replace(":", "");
+    }
```

Comment:
So we can run a checkMsgForStatus(message) to update the printer status inside this function, it's make more sense that let this code inside this function. 
It's better to add this last comment as a code comment to this lines.
```
the returned string from the printer is X:### Y:### Z:### E:#### the above is removing the E section since we do not care about the extruders position. Then we remove the remaining ":"
```